### PR TITLE
Add workflows documentation page

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #395

### CODE REVIEW

**Comment by mdellweg** [comment]
Actually, i am concerned this level of detail may be too much (both for digesting and for keeping up to date).
What i really like about this is that we list all the workflow links for plugin specific docs, as well as the "use `--help` to find yourself around". The latter has the benefit that it is always accurate to the users installation.

File: docs/supported_workflows.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+
+## pulp_ansible
+
+Ansible commands require minimum version of 0.7.0. The commands can be found under the `ansible`
+subgroup.
+
+### Workflows
+
+The CLI currently supports the following workflows:
```
Comment (ggainey) [inline docs/supported_workflows.md:44]: Same here as above - in fact, all the Workflows sections. I think they need a blank line before the lists?

**Review by ggainey — CHANGES_REQUESTED** [review]
Would still like to see the CRUD indicators in bold, as well as fixing the Workflows-lists.

**Comment by gerrod3** [comment]
> Would still like to see the CRUD indicators in bold, as well as fixing the Workflows-lists.

My bad, I wasn't thinking of the CRUD indicators as verbs. Should the letters in the keys also be bolded?

**Comment by ggainey** [comment]
> > Would still like to see the CRUD indicators in bold, as well as fixing the Workflows-lists.
> 
> My bad, I wasn't thinking of the CRUD indicators as verbs. Should the letters in the keys also be bolded?

Hm - I'm not sure we need to. What I'm mostly after here, is to pull the eye to the end of the line and separate the "these are implemented" from "this is the object" - does that make sense?

**Review by ggainey — APPROVED** [review]
Changes look great, thanks!

**Comment by fkuep** [comment]
Thanks @gerrod3 !!
Looking over the document, I can see links to the workflows in the plugins documentation , which most times use the http binary instead of pulp-cli. 
Since the server plugins are out-of-tree of pulp-cli pulp-cli/tree/main/tests/scripts serve as integration tests right ?
Should the tests like tests/scripts/pulp_ansible/test_content.sh , have a comment like : * `docs/supported_workflows.md#pulp_ansible : Role content / Collection content` 
to track what is really known to work, with a certain release ? 


@mdellweg with the addition of `docs/supported_workflows.md` now has *authorative ,versioned information on what it can do* ...
does @melcorr know already she can point to it from https://github.com/pulp/pulpproject.org/blob/gh-pages/content-plugins.md#pulp-3-cli-coverage ?

**Comment by gerrod3** [comment]
> Thanks @gerrod3 !! Looking over the document, I can see links to the workflows in the plugins documentation , which most times use the http binary instead of pulp-cli. Since the server plugins are out-of-tree of pulp-cli pulp-cli/tree/main/tests/scripts serve as integration tests right ? Should the tests like tests/scripts/pulp_ansible/test_content.sh , have a comment like : `docs/supported_workflows.md#pulp_ansible : Role content / Collection content` to track what is really known to work, with a certain release ?

Yes it is up to the plugin writers to change their workflows to use the CLI. Currently `pulp_ansible`, `pulp_file`, and `pulp_python`, have CLI workflows. Some of the `pulpcore` workflows have been converted, but none of the `pulp_container` or `pulp_rpm` have been. It's my goal to work on converting those last 2 plugins next. The tests are currently used as integration tests if they have [`test_cli` set](https://github.com/pulp/pulp_rpm/blob/master/template_config.yml#L69). The tests don't have much visibility to regular users, so I'm not sure how helpful putting comments in them will be.

**Comment by fkuep** [comment]
> . The tests are currently used as integration tests if they have [`test_cli` set](https://github.com/pulp/pulp_rpm/blob/master/template_config.yml#L69).

Yes - OK - Great!
.. thats way better !

**Pull request merged**